meshpy/four_c/dbc_monitor.py

```diff
@@ -88,44 +153,203 @@ def read_dbc_monitor_file(file_path):
     return nodes, data[:, 1], data[:, 4:7], data[:, 7:]
 
 
-def dbc_monitor_to_input(input_file, file_path, step=-1, function=1, n_dof=3):
+def add_point_neuman_condition_to_input_file(
+    input_file: InputFile,
+    nodes: list[int],
+    function_array: list[Function],
+    force: np.ndarray,
+    *,
+    n_dof: int = 3,
+):
     """
-    Convert the Dirichlet boundary condition monitor log to a Neumann
-    boundary condition input section.
+    Adds a Neumann boundary condition to the input file for the given node_ids
+    with the function_array and force values by creating a new geometry set.
 
     Args
     ----
     input_file: InputFile
-        The input file where the created Neumann boundary condition is added
-        to. The nodes referred to in the log file have to match with the ones
-        in the input section. It is advisable to only call this function once
-        all nodes have been added to the input file.
-    file_path: str
-        Path to the Dirichlet boundary condition log file.
-    step: int
-        Step values to be used. Default is -1, i.e. the last step.
-    function: Function, int
-        Function for the Neumann boundary condition.
+        InputFile where the boundary conditions are added to
+    nodes: [node_id]
+        list containing the ids of the nodes for the condition
+    function_array: [function]
+        list with functions
+    force: [np.ndarray]
+        values to scale the function array with
     n_dof: int
         Number of DOFs per node.
     """
 
-    nodes, _, force, _ = read_dbc_monitor_file(file_path)
+    # Add the function to the input file, if they are not previously added.
+    for function in function_array:
+        if function not in input_file.functions:
+            input_file.add(function)
```

Comment:
```suggestion
    input_file.add(function_array)
```
The check is performed under the hood in `Mesh.add`.